src/cockpit/channels/dbus.py

```diff
@@ -225,18 +223,28 @@ def do_open(self, options):
         self.tasks = set()
 
         bus = options.get('bus')
+        address = options.get('address')
 
-        if bus == 'internal':
+        if bus == 'none':
+            if address is None or address == "internal":
+                logger.debug('get internal bus for %s', self.name)
+                self.bus = self.router.internal_bus.client
+            else:
+                logger.debug('get bus with address %s for %s', address, self.name)
+                self.bus = Bus.new(address=address, bus_client=self.name is not None)
```

Comment:
What happens if both address and bus are given?

Reply:
bus must be "none" (the string), otherwise "address" is ignored.

Reply:
This seems annoying.  One would expect that having an unspecified bus but a specified address would result in the address being used, but I guess in that case, we should actually choose the system bus.

I think I'd prefer to see

 - a ChannelError in case address and bus are both specified and bus is anything other than "none"
 - connecting via address in case bus is unspecified (as well as the case where it's the string "none")

Reply:
Yes, I like this.  Note also that `{ bus: "none", address: "...", name: "..." }` will actually make a connection to a bus, so insisting on `bus: "none"` in this case doesn't make sense.